### PR TITLE
Add small refactorings

### PR DESCRIPTION
**Pull request by tranzystorek-io**
Summary:

- Remove some redundant `.iter()`/`.into_iter()` calls
- Reduce nesting in `if-else` blocks that can be easily short circuited
- Use deref coercion to avoid some `.as_...()` calls

### CODE REVIEW

**Comment by Enselic** [comment]
Thanks for taking the time to make a contribution!

May I ask if you could please split the change up into smaller commits? The smaller the better, essentially. If there turns out to be a regression, smaller commits will give higher resolution for both git bisect and git revert. They also make code review easier.

I'm also curious if you used clippy to find these, or if you manually found them, if you don't mind me asking?

**Comment by tranzystorek-io** [comment]
Hi, thanks for having me! 😄

I'll split the change in a bit.

> I'm also curious if you used clippy to find these, or if you manually found them, if you don't mind me asking?

I scanned the code for common idioms manually, e.g. `.iter` or `.as_`, I also manually looked through all the `if` keywords. Sure wish clippy was this smart 😅

File: src/assets.rs
Comment (tranzystorek-io) [reply src/assets.rs]: I see the point you're making, and I agree 😃

Reverted

File: src/bin/bat/app.rs
Comment (tranzystorek-io) [reply src/bin/bat/app.rs]: Reverted, I don't want to refactor code into being inconsistent

**Review by Enselic — APPROVED** [review]
This looks good to me now. Thanks!

I'll give other maintainers a chance to take a look before I merge.

**Review by keith-hall — APPROVED** [review]

**Comment by Enselic** [comment]
Thanks for taking a look, Keith. Let's merge this now to not delay #951.

**Pull request merged**
